kernel/smp.c

```diff
@@ -12,6 +12,7 @@
 
 #ifdef CONFIG_SMP
 static atomic_t global_lock;
+static atomic_t start_flag;
```

Comment:
I pulled in this patch for my own branch and it failed some sanitychecks. This is defined within CONFIG_SMP, but the usage inside smp_init() below is not inside a #ifdef. So there were errors.

Reply:
Seems like it is to me and it builds on both SMP=y/n in CI.  Can you post the error?

Reply:
> zephyr/kernel/smp.c: In function ‘smp_init’:
> zephyr/kernel/smp.c:96: error: ‘start_flag’ undeclared (first use in this function)
> zephyr/kernel/smp.c:96: error: (Each undeclared identifier is reported only once
> zephyr/kernel/smp.c:96: error: for each function it appears in.)
> make[2]: *** [zephyr/kernel/CMakeFiles/kernel.dir/build.make:310: zephyr/kernel/CMakeFiles/kernel.dir/smp.c.obj] Error 2

I was just building `samples/hello_world` with no modification. But this was under XCC.